### PR TITLE
Merge multi-precision support into main

### PR DESCRIPTION
**Pull request by rouson**
This PR is will be ready to merge either when `gfortran`'s support for parameterized derived types (PDTs) has been fixed or we have decided to drop support for `gfortran`.

The `develop` branch that this PR will merge into `main` resulted from merging three significant development branches:

1. The `parallel` branch added support for automatic parallelization of training via `do concurrent`.
2. The `parameterized-derived-types` branch added the ability to support both default-`real` and `double precision` inference without requiring recompiling.
3. The `rename-inference-engine_t` branch, which globally renames the project from "Inference-Engine" to Fiats and correspondingly renames numerous program entities (files, modules, procedures, documentation, etc.)

The latter three branches have been deleted in order to clarify the path forward.  Starting now, new work should branch off of `develop`.

### CODE REVIEW

**Comment by rouson** [comment]
Based on correspondences with `gfortran` developers it appears that the fixes to `gfortran` that would be required for this repository to continue supporting `gfortran` are likely to be a ways out.  Now that the `develop` branch also includes a merge of PR #218, delaying merging this PR would also delay the renaming of Inference-Engine to Fiats.  The name change is somewhat urgent in the sense that this repository is now being featured in several important documents, including anticipated third-party submissions by outside parties in response to the [FASST RFI] and the [ORNL/DOE "New Frontiers" RFP].  It therefore makes sense to merge this PR post haste.

[FASST RFI]: https://www.federalregister.gov/documents/2024/09/12/2024-20676/notice-of-request-for-information-rfi-on-frontiers-in-ai-for-science-security-and-technology-fasst
[ORNL/DOE "New Frontiers" RFP]: https://www.ornl.gov/news/doe-ornl-announce-opportunity-define-future-high-performance-computing

File: README.md [hunk 1]
```diff
@@ -61,26 +59,22 @@ The demonstration applications
...
-#### LLVM (`flang-new`)
-Building with `flang-new` requires passing flags to enable the compiler's experimental support for assumed-rank entities:
 ```
 fpm test --compiler flang-new --flag "-mmlir -allow-assumed-rank -O3"
 ```
 
-##### _Experimental:_ Automatic parallelization of `do concurrent` on CPUs
-With the `amd_trunk_dev` branch of the [ROCm fork] fork of LLVM, automatic parallelization currently works for inference, e.g.
+###### _Experimental:_ Automatic parallelization of `do concurrent` on CPUs
+With the `amd_trunk_dev` branch of the [ROCm fork] fork of LLVM, automatically parallelize inference calculations inside `do concurrent` constructs:
```
Comment (ktras) [inline README.md:77]:
```suggestion
With the `amd-trunk-dev` branch of the [ROCm fork] of LLVM, automatically parallelize inference calculations inside `do concurrent` constructs:
```

File: README.md [hunk 2]
```diff
@@ -89,25 +83,46 @@ fpm run \
...
+
+##### GNU (`gfortran`)
+Compiler bugs related to parameterized derived types currently prevent `gfortran` from building Fiats versions 0.15.0 or later.
+Test and build earlier versions of Fiats build with the following command:
+```
+fpm test --compiler gfortran --profile release
+```
+
+##### Intel (`ifx`)
+Compiler bugs related to parameterized derived types currently prevent `gfortran` from building Fiats versions 0.15.0 or later.
```
Comment (ktras) [inline README.md:108]: This line reference gfortran but is talking about ifx. Is the below suggestion sufficient to be accurate to the current situation?

```suggestion
Compiler bugs related to parameterized derived types currently prevent `ifx` from building Fiats versions 0.15.0 or later.
```

File: README.md [hunk 3]
```diff
@@ -89,25 +83,46 @@ fpm run \
...
+fpm test --compiler ifx --profile release --flag "-fopenmp-target-do-concurrent -qopenmp -fopenmp-targets=spir64 -O3"
 ```
 
-#### HPE (`crayftn.sh`) -- under development
-Support for the Cray Compiler Environment (CCE) Fortran compiler is under development.
-Building with the CCE `ftn` compiler wrapper requires an additional trivial wrapper
-shell script. For example, create a file `crayftn.sh` with the following contents and
-place this file's location in your `PATH`:
+#### Under Development
+We are corresponding with the developers of the compiler(s) below developers about addressing the compiler issues preventing building newer Fiats releases.
```
Comment (ktras) [inline README.md:121]:
```suggestion
We are corresponding with the developers of the compiler(s) below about addressing the compiler issues preventing building Fiats.
```

File: README.md (hunk 2 above)
Comment (rouson) [reply README.md:108]:
```suggestion
Compiler bugs related to generic name resolution currently prevent `ifx` from building Fiats versions 0.15.0 or later.
```

**Pull request merged**